### PR TITLE
ath79: extra gmac configurations from DT on qca956x

### PR DESCRIPTION
**Pull request by j-d-r**
Here are two patches:

- first: add extra registers and masks for GMAC of SoCs qca956x.
Values are taken from [CAF](https://source.codeaurora.org/quic/qsdk/oss/boot/u-boot-1.1.4/tree/include/956x.h).
- second: import values from device tree

Regards

### CODE REVIEW

**Comment by juppin** [comment]
As long as no one uses it, this is almost useless...

**Comment by CodeFetch** [comment]
@juppin It's good for completeness sake and if someone needs it at some point it's already there.

**Review by CodeFetch — APPROVED** [review]
What is CAF U-Boot? Can you give a link for reference?

**Comment by j-d-r** [comment]
Feel free to do whatever you want of these two commits.

I'm using some of theses settings in my support for TP-Link EAP 245v1 https://github.com/j-d-r/openwrt/commit/c59e1665caae99f4a18f1328c2accac4524aca2e as TP-Link uses them in their GPL code (Linux 3.3.8)

CAF stands for code aurora forum. Qualcomm publishes here their open source code.

U-Boot 1.1.4 for AP152 - QCA956x reference board - is published here. Header https://source.codeaurora.org/quic/qsdk/oss/boot/u-boot-1.1.4/tree/include/956x.h contains most registers definitions.

**Comment by blocktrron** [comment]
@j-d-r Have you already sent the patch you are adding here to linux-mips?

**Comment by j-d-r** [comment]
@blocktrron No I didn't, kernel patch is only some #define.

**Comment by adrianschmutzler** [comment]
@j-d-r "some" is an understatement

File: target/linux/ath79/patches-4.14/920-qca956x-add-more-registers.patch [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
++#define QCA956X_PLL_PCIE_PLL_DITHER_DIV_MAX_REG		0x18
++#define QCA956X_PLL_PCIE_PLL_DITHER_DIV_MIN_REG		0x1c
++#define QCA956X_PLL_SWITCH_CLOCK_SPARE_REG		0x28
++#define QCA956X_PLL_CURRENT_PCIE_PLL_DITHER_REG		0x2c
++#define QCA956X_PLL_ETH_XMII_REG			0x30
++#define QCA956X_PLL_BB_PLL_CONFIG_REG			0x34
++#define QCA956X_PLL_DDR_PLL_DITHER1_REG			0x38
++#define QCA956X_PLL_DDR_PLL_DITHER2_REG			0x3c
++#define QCA956X_PLL_CPU_PLL_DITHER1_REG			0x40
++#define QCA956X_PLL_CPU_PLL_DITHER2_REG			0x44
```
Comment (CodeFetch) [inline target/linux/ath79/patches-4.14/920-qca956x-add-more-registers.patch]: Indentation issues. I think this should be sent upstream, too.

**Review by CodeFetch — CHANGES_REQUESTED** [review]

File: target/linux/ath79/patches-4.19/920-qca956x-add-more-registers.patch [hunk 2]
```diff
@@ -0,0 +1,191 @@
+MIPS: ath79: add lots of missing registers for QCA956x SoCs
+
+This patch adds many registers and values for QCA956x GMAC interface.
+
+QCA956X_PLL_SWITCH_CLOCK_CONTROL_REG, QCA956X_PLL_ETH_XMII_CONTROL_REG
+and QCA956X_ETH_CFG_RGMII_EN names are fixed to match original names.
+
+Values are taken from U-Boot 1.1.4 of CAF.
```
Comment (CodeFetch) [inline target/linux/ath79/patches-4.19/920-qca956x-add-more-registers.patch]: What is CAF?

**Comment by adrianschmutzler** [comment]
@blocktrron @981213 Would one of you provide a verdict on these changes?

From the comments, it looks to me like we don't need them right now, so no reason to care for now?

**Comment by blocktrron** [comment]
Dragonfly as well as Jet platforms do not offer MII interfaces other than SGMII to my knowledge. So all settings except for the MII mux (ge0-sgmii) seem to be superfluous to me.

However (in case this multiplexer exists) and it's configured incorrectly, we should configure it for this platform static to ge0, as the QCA9563 only has one MAC. But as this is not an issue (as far as I'm concerned about this PR).

I'm for closing here. If we need any of those knobs we can add them later. I'm not to fond about dead code in a driver we are currently maintaining ourselves.

**Pull request closed without merging**